src/cljs/org/broadinstitute/firecloud_ui/page/library/library_page.cljs

```diff
@@ -97,24 +98,38 @@
                                              "Please contact " [:a {:target "_blank" :href (str "mailto:" (:library:contactEmail data))} (str (:library:datasetCustodian data) " <" (:library:contactEmail data) ">")]
                                                " and request access for the "
                                                (:namespace data) "/" (:name data) " workspace."])})))}))
+   :build-aggregate-fields
+   (fn [{:keys [props]}]
+     (reduce
+     (fn [results field] (assoc results field (if (contains? (:expanded-aggregates props) field ) 0 5)))
+     {}
+   (:aggregate-fields props)))
    :pagination
-   (fn [{:keys [state]}]
-     (fn [{:keys [current-page rows-per-page filter-text]} callback]
-       (endpoints/call-ajax-orch
-         (let [from (* (- current-page 1) rows-per-page)]
-           {:endpoint endpoints/search-datasets
-            :payload {:searchString filter-text :from from :size rows-per-page}
-            :headers utils/content-type=json
-            :on-done
-            (fn [{:keys [success? get-parsed-response status-text]}]
-              (if success?
-                (let [{:keys [total results]} (get-parsed-response)]
-                  (swap! state assoc :total total)
-                  (callback {:group-count total
-                             :filtered-count total
-                             :rows results}))
-                (callback {:error status-text})))}))))})
-
+   (fn [{:keys [this state props]}]
+     (fn [{:keys [current-page rows-per-page]} callback]
+       (when-not (empty? (:aggregate-fields props))
+         (endpoints/call-ajax-orch
+           (let [from (* (- current-page 1) rows-per-page)]
+             {:endpoint endpoints/search-datasets
+              :payload {:searchString (:search-text props)
+                        :filters (utils/map-kv (fn [k v]
+                                                 [(name k) v])
+                                               (:facet-filters props))
+                        :from from
+                        :size rows-per-page
+                        :fieldAggregations (if (= 1 current-page) (react/call :build-aggregate-fields this) {})}
```

Comment:
is this an optimization so we don't re-request aggregates when we are paginating?